integration/src/main/java/scripts/utils/NotebookUtils.java

```diff
@@ -196,16 +199,19 @@ public static void assertInstanceHasProxyUrl(
                         .execute()
                         .getProxyUri();
                 if (p == null) {
+                  logger.info("Fails to fetch notebook proxy url, retry");
                   throw new NullPointerException();
                 }
                 return p;
                 // Retry 403s as permissions may take time to propagate, but do not retry if it's
                 // any other IO exception.
               } catch (GoogleJsonResponseException e) {
                 if (e.getStatusCode() == HttpStatus.SC_FORBIDDEN) {
+                  logger.info("Fails to fetch notebook proxy url due to 403, retry", e);
                   throw e;
                 }
               } catch (IOException ignored) {
+                logger.info("Fails to fetch notebook proxy url, do not retry", ignored);
               }
               return null;
```

Comment:
There are two cases that are not logged with this code change:
1. When the GoogleJsonResponseException is not Forbidden
2. When the exception is not GoogleJsonReponseException or IOException
I would like logging of those cases.

Reply:
Ah, I see. Great, thanks!